src/TSTransformer/nodes/expressions/transformYieldExpression.ts

```diff
@@ -12,14 +12,30 @@ export function transformYieldExpression(state: TransformState, node: ts.YieldEx
 	const expression = transformExpression(state, node.expression);
 	if (node.asteriskToken) {
 		const loopId = luau.tempId("result");
+		const returnValue = luau.tempId("return");
+
+		state.prereq(
+			luau.create(luau.SyntaxKind.VariableDeclaration, {
+				left: returnValue,
+				right: undefined,
+			}),
+		);
```

Comment:
```suggestion
		const returnValue = state.pushToVar(undefined, "returnValue");
```
This prereq can be simplified by `pushToVar`ing `undefined`.

I'm also suggesting changing the used name, because although `_return` is technically valid, it feels "wrong" to use a language keyword for the variable.